test/test_macros.h

```diff
@@ -56,3 +56,7 @@ bool equals(const T1 &a, const T2 &b, const Tol tolerance) {
  * functions */
 #define ASSERT_NO_THROW_NODISCARD(expr) ASSERT_NO_THROW((void)expr)
 #define ASSERT_THROW_NODISCARD(expr, type) ASSERT_THROW((void)expr, type)
+
+// Helper that casts the result of `statement` to void to suppress warnings.
+#define EXPECT_THROW_DROPRES(statement, expected_expression)                   \
```

Comment:
1. `EXPECT_THROW_DROPRES` -> `EXPECT_THROW_DROP_RES`?

2. Should this be `[[maybe_unused]] auto var = statement`?, slight tangent - elsewhere I have used old c-style casts to remove unused warnings rather than `static_cast`, so I should update those.

3. I think we should either use this macro everywhere, or not at all rather than the mix of approaches currently used.

Reply:
>     1. `EXPECT_THROW_DROPRES` -> `EXPECT_THROW_DROP_RES`?
Done

>     2. Should this be `[[maybe_unused]] auto var = statement`?
Not sure if the assignment / copy / move construction would imply an extra condition on the returned type. (Probably irrelevant as we don't have any types that can only appear as rvalue references.)
We would have to put the assignment into a nested scope to avoid name clashes with the test case. But even then, we might have to generate a variable name automatically in order to safely avoid a shadowing warning.

>     3. I think we should either use this macro everywhere, or not at all rather than the mix of approaches currently used.
Done (I think)